### DISCUSSION TITLE
Constantly getting the error: MCMCSamplingError

### DISCUSSION THREAD

**Discussion by Rupinder-Singh-Bell** [google/meridian#367] [Q&A] [answered]
I get the error "MCMCSamplingError: MCMC sampling failed with a maximum R-hat value of 2220934037504.0" or other very large values when i run "model_diagnostics.plot_rhat_boxplot()" even though the model runs.

 I have tried increasing the parameters in the meridian.sample_posterior to meridian.sample_posterior(n_chains=7, n_adapt=2000, n_burnin=2000, n_keep=4000) but still get this error. 

My data has 2 geographies and 2 years of weekly data with 12 media channels and only controls are GQV and Price.

Any suggestions to fix this?

**Comment by gauravsab1** [comment]
It appears your MCMC sampler is [not converging](https://developers.google.com/meridian/docs/user-guide/model-diagnostics?hl=en#assess-convergence). Here are a few steps you can take to remedy the issue:

1. Check if there is high correlation among your explanatory variables (e.g., correlation between two different medias). See [Checking correlation between variables](https://developers.google.com/meridian/docs/user-guide/perform-eda?hl=en) on this page. 
2. Check if one or more variables have too many zeros. 
3. [Reduce](https://developers.google.com/meridian/docs/user-guide/configure-model?hl=en#tune-automatic-seasonality) the number of knots (for example, you could use a knot for each month instead of each week). 
4. Lastly, you could restrict coefficients to be the same across geos, since you have only two geos (though you may not wish to do this if you specifically defined the two geos to be quite different from one another). For instance, set [`eta_m`](https://developers.google.com/meridian/docs/advanced-modeling/default-prior-distributions?hl=en#eta_m_and_eta_rf) and [`xi_c`](https://developers.google.com/meridian/docs/advanced-modeling/default-prior-distributions?hl=en#xi_c) to be `tfp.distributions.Deterministic(0)` in the prior distributions object (see [here](https://developers.google.com/meridian/docs/user-guide/configure-model?hl=en#set-priors) how to adjust priors).

**Reply by Rupinder-Singh-Bell** [reply, marked as answer]
The issue was there was a media channel that had 0 spend and impressions which after excluding this media channel from the model it was able work when i ran the "model_diagnostics.plot_rhat_boxplot()". The model spec and data loader all worked fine and even the meridian.sample_posterior() functions all worked normally so a suggestion would be if Meridian in the data loader could raise an error if it has a NaNs to prevent this type of issue in the future.
